src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -539,5 +539,38 @@ public static int Count<T>(this ImmutableArray<T> items, Func<T, bool> predicate
 
             return dictionary;
         }
+
+        /// <summary>
+        /// Checks whether two arrays of elements who are IEquatable are equal.
```

Comment:
Please document explicitly that it treats empty and default as distinct values. I'd almost prefer if we renamed the function to be explicit to that point since it's probably a mix if you want that behavior or not.